app/models/queen.rb

```diff
@@ -1,4 +1,15 @@
 class Queen < Piece
+
+  def valid_move?(destination_x, destination_y)
+    valid = super(destination_x, destination_y)
+    horizontal = (destination_y == self.y_position)
```

Comment:
Horizontal, vertical, and diag is implemented in https://github.com/the-cheaters/up-all-knight/blob/master/app/models/piece.rb#L48 - maybe make a helper. Although it would be best to submit just the helper in another PR so that team is aware and can use in their moves too without having to wait on this merge.